src/main/java/gregtech/common/metatileentities/converter/ConverterTrait.java

```diff
@@ -0,0 +1,268 @@
+package gregtech.common.metatileentities.converter;
+
+import gregtech.api.GTValues;
+import gregtech.api.capability.FeCompat;
+import gregtech.api.capability.GregtechCapabilities;
+import gregtech.api.capability.IEnergyContainer;
+import gregtech.api.metatileentity.MTETrait;
+import gregtech.api.util.GTUtility;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.energy.CapabilityEnergy;
+import net.minecraftforge.energy.IEnergyStorage;
+
+public class ConverterTrait extends MTETrait {
```

Comment:
Why does this not extend `EnergyContainerHandler`?

Reply:
I think this could be refactored to extend that class but I don't think it needs to happen immediately

Reply:
#687